ts/store/reducers/deferred-navigation.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * A reduced for deferred navigation actions.
+ */
+
+import {
+  CLEAR_DEFERRED_NAVIGATION_ACTION,
+  SAVE_NAVIGATION_STATE,
+  SAVE_NAVIGATION_WHEN_LOGGED_IN
+} from "../actions/constants";
+import { Action } from "../actions/types";
+import { GlobalState } from "./types";
+
+export interface DeferredNavigationActionState {
+  // Navigation saved before going to background.
+  navigationState: Action | undefined;
+  navigation: Action | undefined;
```

Comment:
explain the difference between `navigationState` and `navigation` and what's stored in them